raycicmd/command_step.go

```diff
@@ -74,6 +74,14 @@ func (c *commandConverter) convert(id string, step map[string]any) (
 	if err != nil {
 		return nil, fmt.Errorf("map agent: %w", err)
 	}
+	concurrencyGroup, _ := stringInMap(step, "concurrency_group")
+	restrictedConcurrencyGroupPrefixes := c.config.RestrictedConcurrencyGroupPrefixes
+	if concurrencyGroup != "" && restrictedConcurrencyGroupPrefixes != nil {
```

Comment:
you might want to check `len(restrictedConcurrencyGroupPrefixes) > 0` here. as it is quite hard to distinguish `nil` and `[]string{}` in yaml files.

Reply:
ok. that works too. could you add a comment here to make it clear that we treat nil and empty slice differently? in go, it is often treated the same in most parts of the common logic.

it is kind of like in python if one write `if list:` then it needs to be not None, also needs to be not an empty list.